### PR TITLE
Add swagger rest api

### PR DESCRIPTION
**Pull request by dgkncelik**
Swagger is a documentation tool for Rest API services. After running application on localhost
http://localhost:8080/swagger-ui.html address will give the swagger output.

### CODE REVIEW

**Comment by karakayasemi** [comment]
what is missing in here?

File: src/main/java/edu/itu/cavabunga/business/CalendarManagerServiceImpl.java [hunk 1]
```diff
@@ -54,6 +54,10 @@ public Participant getParticipantByUserName(String userName){
 
     @Override
     public List<Participant> getAllParticipants(){
+        if(participantService.getAllParticipant().isEmpty()){
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/business/CalendarManagerServiceImpl.java]: no need to this addition in here

File: src/main/java/edu/itu/cavabunga/controller/ComponentController.java [hunk 2]
```diff
@@ -20,37 +26,78 @@ public ComponentController(CalendarManagerService calendarManagerService) {
...
     }
 
-    @DeleteMapping("/{component_id}/")
+    @ApiOperation(value = "Delete component with id = {component_id}")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "Successfully delete component with id = {component_id}", response = Response.class),
+            @ApiResponse(code = 400, message = "Bad request", response = ErrorResponse.class),
+            @ApiResponse(code = 401, message = "Unauthorized request", response = ErrorResponse.class),
+            @ApiResponse(code = 403, message = "Forbidden", response = ErrorResponse.class),
+            @ApiResponse(code = 404, message = "No Wcomponent found with id = {component_id}", response = ErrorResponse.class),
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/controller/ComponentController.java]: typo

File: src/main/java/edu/itu/cavabunga/controller/ParticipantController.java [hunk 3]
```diff
@@ -18,33 +24,73 @@ public ParticipantController(CalendarManagerService calendarManagerService) {
         this.calendarManagerService = calendarManagerService;
     }
 
-    @GetMapping
+    @ApiOperation(value ="View all participants available", response = Iterable.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "Successfully retrieved list of participants",response = ParticipantResponse.class),
+            @ApiResponse(code = 401, message = "Unauthorized request", response = ErrorResponse.class),
+            @ApiResponse(code = 403, message = "Forbidden", response = ErrorResponse.class),
+            @ApiResponse(code = 404, message = "No partipants found", response = ErrorResponse.class),
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/controller/ParticipantController.java]: pls remove

File: src/main/java/edu/itu/cavabunga/controller/PropertyController.java [hunk 4]
```diff
@@ -20,35 +26,75 @@ public PropertyController(CalendarManagerService calendarManagerService) {
...
     @DeleteMapping("/{property_id}")
     @ResponseStatus(HttpStatus.OK)
     public Response deleteProperty(@PathVariable(value = "property_id") Long propertyId) {
         calendarManagerService.deleteProperty(propertyId);
         return new Response(0,"deleted");
     }
 
+    @ApiOperation(value = "Get parameters of property with id = {property_id}")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "Succefully retrieve parameters of property with id = {property_id}", response = ParameterResponse.class),
```
Comment (karakayasemi) [inline src/main/java/edu/itu/cavabunga/controller/PropertyController.java]: typo

**Review by karakayasemi — CHANGES_REQUESTED** [review]
The code looks okay for me after reviewed small changes, how can I test and see output HTML?

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -1,3 +1,3 @@
 spring.jpa.hibernate.ddl-auto=create
-spring.datasource.url=jdbc:mysql://127.0.0.1/test
```
Comment (karakayasemi) [inline src/main/resources/application.properties]: revert

**Review by karakayasemi — CHANGES_REQUESTED** [review]

**Comment by karakayasemi** [comment]
@gitmate rebase

**Comment by karakayasemi** [comment]
@gitmate-bot rebase

**Comment by karakayasemi** [comment]
@gitmate-bot rebase

**Comment by karakayasemi** [comment]
@gitmate-bot rebase

**Comment by itudevops** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by itudevops** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by karakayasemi** [comment]
@dgkncelik please squash your last 2 commits, they are irrelevant for development tree.

**Review by karakayasemi — APPROVED** [review]

**Pull request merged**